src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameWriterV2Tests.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Spark.E2ETest.Utils;
+using Microsoft.Spark.Sql;
+using Microsoft.Spark.UnitTest.TestUtils;
+using Xunit;
+
+namespace Microsoft.Spark.E2ETest.IpcTests
+{
+    [Collection("Spark E2E Tests")]
+    public class DataFrameWriterV2Tests
+    {
+        private readonly SparkSession _spark;
+
+        public DataFrameWriterV2Tests(SparkFixture fixture)
+        {
+            _spark = fixture.Spark;
+        }
+
+        /// <summary>
+        /// Test signatures for APIs introduced in Spark 3.*.
+        /// </summary>
+        [SkipIfSparkVersionIsLessThan(Versions.V3_0_0)]
+        public void TestSignaturesV3_0_X()
+        {
+            DataFrame df = _spark
+                .Read()
+                .Schema("age INT, name STRING")
+                .Json($"{TestEnvironment.ResourceDirectory}people.json");
+
+            DataFrameWriterV2 dfwV2 = df.WriteTo("testtable");
+
+            Assert.IsType<DataFrameWriterV2>(dfwV2.Using("json"));
+
+            Assert.IsType<DataFrameWriterV2>(dfwV2.Option("key1", "value"));
+            Assert.IsType<DataFrameWriterV2>(dfwV2.Option("key2", true));
+            Assert.IsType<DataFrameWriterV2>(dfwV2.Option("key3", 1L));
+            Assert.IsType<DataFrameWriterV2>(dfwV2.Option("key4", 2D));
+
+            Assert.IsType<DataFrameWriterV2>(dfwV2.Options(
+                new Dictionary<string, string>() { { "key", "value" } }));
+
+            Assert.IsType<DataFrameWriterV2>(dfwV2.TableProperty("prop", "value"));
+
+            _spark.Sql("DROP TABLE IF EXISTS default.testtable");
+            dfwV2.Create();
+
+            Assert.IsType<DataFrameWriterV2>(dfwV2.PartitionedBy(df.Col("age")));
+
+
+            // Testing caveat *************************************************************
+            // The following functions cannot be tested because of lack of support for DataSourceV2.
+            // This is because Spark 3.0 currently doesn't support file source as provider for
+            // tables. Issue - https://issues.apache.org/jira/browse/SPARK-28396
+            //
+            // 1. DataFrameWriterV2.Replace()
```

Comment:
Spark uses an in memory catalog for the tables.  Is there any way we can achieve the same ?

Reply:
Can you set `spark.sql.sources.useV1SourceList` to "" and see if it works? Spark 3.0 comes with datasource v2 implementation for the built-in sources (json, parquet, etc.), but it's disabled by default.

Reply:
Still same error: `org.apache.spark.sql.AnalysisException: REPLACE TABLE AS SELECT is only supported with v2 tables.`

Reply:
Oops. yea, this requires v2 catalog, not datasources v2 (confused by the comment)

Reply:
There is no v2 catalog implementation out of box (one in test but not a part of the jars shipped).

Instead of commenting these out, can we catch an exception and put the snippet of the exception message as comment? (I think we did something similar for not supported udf types).

Reply:
Gotcha. Yes will do.